walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,55 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<Event> existingEvents = new ArrayList<Event>(events);
+    List<TimeRange> possibleSchedule = new ArrayList<TimeRange>();
+
+    Collections.sort(existingEvents);
+    int curStartTime = 0;
+    TimeRange eventTimeRange;
+    boolean contains;
+
+    for (Event event : existingEvents) {
+      contains = false;
+      for (String participant : request.getAttendees()) {
```

Comment:
Alternatively you can also use `Collections.disjoint` to do this.

Reply:
Replaced code with Collections.disjoint